File: dgis_maps_flutter_android/android/src/main/kotlin/pro/flown/dgis_maps_flutter_android/DgisMapController.java
package pro.flown.dgis_maps_flutter_android;

import android.content.Context;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.annotation.VisibleForTesting;

import io.flutter.plugin.common.BinaryMessenger;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.platform.PlatformView;
import kotlin.Unit;
import kotlin.jvm.functions.Function1;
import ru.dgis.sdk.map.Map;
import ru.dgis.sdk.map.MapView;

public class DgisMapController implements PlatformView {
    private final int id;
    private final MethodChannel methodChannel;
    @Nullable
    private MapView mapView;
    private Map dgisMap;
    private final Context context;

    DgisMapController(
            int id,
            Context context,
            BinaryMessenger binaryMessenger) {
        this.id = id;
        this.context = context;
        this.mapView = new MapView(context);
        methodChannel = new MethodChannel(binaryMessenger, "dgis_maps_flutter_" + id);
    }

    @Override
    public View getView() {
        return mapView;
    }

    @Override
    public void dispose() {
        methodChannel.setMethodCallHandler(null);
    }

    void init() {
        mapView.getMapAsync((map) -> {
            dgisMap = map;
            return null;
        });
    }

}

File: dgis_maps_flutter_android/android/src/main/kotlin/pro/flown/dgis_maps_flutter_android/DgisMapController.kt
package pro.flown.dgis_maps_flutter_android

import android.content.Context
import android.util.Log
import android.view.View
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.platform.PlatformView
import ru.dgis.sdk.coordinates.GeoPoint
import ru.dgis.sdk.map.*
import ru.dgis.sdk.map.Map

class DgisMapController internal constructor(
    id: Int,
    context: Context,
    binaryMessenger: BinaryMessenger?
) : PlatformView, MethodChannel.MethodCallHandler {
    private val methodChannel: MethodChannel =
        MethodChannel(binaryMessenger!!, "dgis_maps_flutter_$id")
    private val mapView: MapView =
        MapView(
            context,
            MapOptions().also {
                it.position = CameraPosition(
                    GeoPoint(30.0, 30.0),
                    Zoom(3f),
                )
            }
        )
    private lateinit var map: Map

    init {
        mapView.getMapAsync { init(it) }
    }

    override fun getView(): View {
        return mapView
    }

    override fun dispose() {
        methodChannel.setMethodCallHandler(null)
    }

    private fun init(map: Map?) {
        if (map == null) return
        this.map = map
        methodChannel.setMethodCallHandler(this)
        methodChannel.invokeMethod("blah", 1234)
        map.camera.stateChannel.connect {
            when(it){
                CameraState.BUSY -> TODO()
                CameraState.FLY -> TODO()
                CameraState.FOLLOW_POSITION -> TODO()
                CameraState.FREE -> TODO()
            }
        }
    }

    fun pubIdle() {

    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        when (call.method) {
            "some" -> {

            }
        }
    }
}